### PR TITLE
Add TimeSeriesOp for time-based data interpolation

### PR DESCRIPTION
**Pull request by chrisgervang**
Introduces the TimeSeriesOp operator to interpolate time series data at a given time using linear interpolation. Includes comprehensive tests for edge cases, field passthrough, and correct interpolation of numeric values.

The API aligns closely to TripsLayerOp, and it's expected they'll typically be used together.

Screenshot 2025-11-11 at 11 42 18 AM

https://github.com/user-attachments/assets/c0c98e82-d40f-438d-83dc-dd36818621b9

### CODE REVIEW

File: noodles-editor/src/noodles/operators.ts [hunk 1]
```diff
@@ -5014,6 +5014,136 @@ export class MaskExtensionOp extends Operator<MaskExtensionOp> {
...
+    return { time: currentTime }
+  }
+
+  // If only one data point, return it
+  if (timeSeries.length === 1) {
+    return { ...timeSeries[0] }
+  }
+
+  // Sort by time to ensure proper interpolation
+  const sortedData = [...timeSeries].sort((a, b) => a.time - b.time)
```
Comment (akre54) [inline noodles-editor/src/noodles/operators.ts]: Is there a way that we can assume this data is sorted, or give you an option to sort it ahead of time? That way we can use a binary search which will be much faster

File: noodles-editor/src/noodles/operators.ts [hunk 2]
```diff
@@ -5014,6 +5014,136 @@ export class MaskExtensionOp extends Operator<MaskExtensionOp> {
...
+  // If currentTime is after last point, return last point
+  if (currentTime >= sortedData[sortedData.length - 1].time) {
+    return { ...sortedData[sortedData.length - 1] }
+  }
+
+  // Find the two points to interpolate between
+  let beforeIndex = 0
+  let afterIndex = 1
+
+  for (let i = 0; i < sortedData.length - 1; i++) {
```
Comment (akre54) [inline noodles-editor/src/noodles/operators.ts]: Can we do a binary search since we know this is already sorted?
